### ISSUE TITLE
Can't go to the static method of a class

### ISSUE THREAD

**Issue by aengusjiang** [microsoft/vscode-cpptools#635]
We can't go to the definition when view a static method of a class, such as:

MainSrvProcessor::clientAskPVEEndReqProcessor;

we can't find the definition of clientAskPVEEndReqProcessor method by press key F12.

please help me to check this and fix it. thanks.

**Comment by sean-mcmanus** [comment]
It works for me. Can you try `class MainSrvProcessor { static void clientAskPVEEndReqProcessor () {} };
int main() { MainSrvProcessor::clientAskPVEEndReqProcessor (); }`

Can you give more repro details? Are other methods of the class found? Does it find the wrong symbol? Maybe something else in the class definition is causing a failure. What OS are you on?

**Comment by aengusjiang** [comment]
Glad for yours response!

I just write a demo(See the attach):
[test.tar.gz](https://github.com/Microsoft/vscode-cpptools/files/923372/test.tar.gz)
My Os is Ubuntu17.04 and 
vs code Version 1.11.2 
cpptools Version 0.10.5

thank you.

**Comment by sean-mcmanus** [comment]
Thanks for the repro -- namespace::class:: works but class:: alone doesn't.

**Comment by keweilizhe** [comment]
yes, i meet this too, can't goto class::my_struct.  (*^__^*) ……

In Expectation your solution,

**Comment by aengusjiang** [comment]
Haha , This issue is fixed in Vscode 1.16.0

**Comment by sean-mcmanus** [comment]
The bug still exists when the IntelliSense engine is set to "Tag Parser". You probably are using the "Default" implementation, which doesn't have the bug. So the bug isn't really fixed, but using the Default intelliSense engine setting is recommended anyway.

**Comment by aengusjiang** [comment]
Hello, sean-mcmanus
I can reproduce the bug again.
I do some change of the the test code above. It maybe can help you to fix this bug.
I guess it is caused by namespace.
see the attachement.
[test_failed.zip](https://github.com/Microsoft/vscode-cpptools/files/1302099/test_failed.zip)
[test_OK.zip](https://github.com/Microsoft/vscode-cpptools/files/1302101/test_OK.zip)

**Comment by sean-mcmanus** [comment]
@AbelaroJiang The bug was never fixed for the Tag Parser. Are you using the "Default" IntelliSense engine (which shouldn't have this bug) or are you able to switch to using that? If you have #include files that aren't found in the translation unit it will fall back to the Tag Parser (if the intelliSenseFallback is enabled).

**Comment by aengusjiang** [comment]
Yes, I use the default IntelliSense engine.
you can download the test sample the compare them.

**Comment by sean-mcmanus** [comment]
Go to Definition is still implemented by the Tag Parser so the bug still existed. When you said it stopped reproing, I got this issue confused with another bug that stops reproing when switching to the Default IntelliSense. I've made a change that fixes this issue, but I'm not sure if it will make the next update or not. FYI, autocomplete and signatureHelp were also broken in this scenario (whenever code was only partially scoped, with some scopes missing on the left). Thanks a lot for reporting this issue. I'm sort of shocked that this scenario has been broken for so long.
